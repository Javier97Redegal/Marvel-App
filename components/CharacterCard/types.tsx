type CharacterCardProps = {
    character: {
        id: number
        name: string | undefined
        thumbnail: {
            path: string
            extension: string
        }
    }
}