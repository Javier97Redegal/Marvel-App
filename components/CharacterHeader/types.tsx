export interface CharacterHeaderProps {
    id: number
}