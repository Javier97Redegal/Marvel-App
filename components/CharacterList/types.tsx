export interface CharacterListProps {
    mode: string
    query: string
}