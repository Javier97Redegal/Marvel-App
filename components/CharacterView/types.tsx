export interface CharacterViewProps {
    id: string
}