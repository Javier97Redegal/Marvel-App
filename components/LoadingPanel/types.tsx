export interface LoadingPanelProps {
    type: 'error' | 'loading' | 'idle' | 'success'
}