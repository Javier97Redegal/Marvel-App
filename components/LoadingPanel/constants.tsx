export const loadingMessage = {
    success: 'No results',
    error: 'API error',
    loading: 'Loading...',
    idle: 'Waiting...'
}