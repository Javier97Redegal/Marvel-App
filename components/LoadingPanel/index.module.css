.loader {
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 2rem;
    padding: 3rem 2rem;

    @media (max-width: 1280px) {
        font-size: 1.5rem;        
    }
}