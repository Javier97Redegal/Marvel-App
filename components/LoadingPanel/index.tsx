import styles from './index.module.css'
import { LoadingPanelProps } from './types'
import { loadingMessage } from './constants'

const LoadingPanel = ({ type }: LoadingPanelProps) => <div className={styles.loader}>{loadingMessage[type]}</div>

export default LoadingPanel