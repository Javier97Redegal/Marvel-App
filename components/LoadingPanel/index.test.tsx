import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import LoadingPanel from '.'
import { loadingMessage } from './constants'
import { LoadingPanelProps } from './types'

describe('LoadingPanel tests', () => {
    const mockProps: LoadingPanelProps = {
        type: 'loading',
    }

    it('displays the correct loading message', () => {
        const { asFragment } = render(<LoadingPanel {...mockProps} />)

        expect(screen.getByText(loadingMessage[mockProps.type])).toBeInTheDocument()
        expect(asFragment()).toMatchSnapshot()
    })
})
