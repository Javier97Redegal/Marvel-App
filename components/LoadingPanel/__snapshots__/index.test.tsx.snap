// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingPanel tests displays the correct loading message 1`] = `
<DocumentFragment>
  <div
    class="loader"
  >
    Loading...
  </div>
</DocumentFragment>
`;
